### PR TITLE
News case study

### PR DESCRIPTION
**Pull request by katejeffreys**

### CODE REVIEW

File: app/1.0/toolbox/news-case-study.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+```html
+<div class="container" fade-in$="[[!loading]]" hidden$="[[failure]]">
+  <article class="flex">
+    <amp-viewer id="ampViewer" src="[[_getArticleSrc(article.id)]]"></amp-viewer>
+  </article>
+```
+
+Note: the AMP version of the app does not support swiping between articles - doing so would require loading the title, image and author of the previous or next article. In the AMP version of News, that data is only loaded when the article content is loaded. It is not loaded with the list of articles.
+
+## More Resources
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: I think to highlight the other docs, this should be, "Docs and more resources" maybe (?) and add a link to the microsite, like, "For documentation on using and customizing the News app ..."

File: app/1.0/toolbox/news-case-study.md [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+      self.response.out.write(open('data/articles/' + kwargs['article_id'] + '.html').read())
+    else:
+      self.response.out.write(open('index.html').read())
+```
+
+## Resource URLs
+
+Resource URLs, like `/bower_components/`, `/data/` and `/src/` are served as static files. These static URLs are defined in `app.yaml`. These URLs are used internally by the app to load components and data.
+
+## Displaying Ads
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: Ads => ads

File: app/1.0/toolbox/news-case-study.md [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+Views in the News app use the same implementation as the Shop app. [See the Shop app case study for more information on views and routing](https://www.polymer-project.org/1.0/toolbox/case-study#views).
+
+For more information on encapsulated routing, see [Encapsulated routing with elements](https://www.polymer-project.org/1.0/blog/routing).
+
+For more information on the `<app-route>` Polymer element used in these implementations, see the [`<app-route>` API reference](https://elements.polymer-project.org/elements/app-route).
+
+There are two main views in the News app. The `<news-list>` element displays a list of articles for the selected category - for example, Top Stories. The `<news-article>` element displays article content. When the user is offline and the category data is not cached, both views display a network error via the `<news-network-warning>` element:
+
+`news-list.html`
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: For these headers, add `{.caption}` on a new line immediately below the filename header to set them off.

```
`news-list.html`
{.caption}
```

File: app/1.0/toolbox/news-case-study.md [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+If the user is offline but category data is cached and ServiceWorker is installed,
+the cached content is displayed.
+
+-   [`<app-route>` API reference](https://elements.polymer-project.org/elements/app-route)
+
+## Routing and Data Binding in the News App
+
+Routing and data bindings work together in the News app to retrieve and display the data relevant to the URL that the user selects.
+
+Polymer's data system allows for data to flow one-way (downward-only, from host element to target element) or two-way (from host to target and target to host). For more information on data binding in Polymer, see [fkjsdkfjl].
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: While fkjsdkfjl is a reputable source on the subject of data binding, I think I'd go for the more canonical `[Data system concepts](../docs/devguide/data-system)` or maybe `[Data binding](../docs/devguide/data-binding)`.

File: app/1.0/toolbox/news-case-study.md [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+
+## Resource URLs
+
+Resource URLs, like `/bower_components/`, `/data/` and `/src/` are served as static files. These static URLs are defined in `app.yaml`. These URLs are used internally by the app to load components and data.
+
+## Displaying Ads
+
+The News app attempts to display ads, without blocking a fast first paint.
+
+The <news-iframe> element uses [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), a web platform API. `IntersectionObserver` is used to determine whether `<news-iframe>` element is visible on screen before creating an iframe that will display an ad.
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: Missing backticks around `<news-iframe>`

File: app/1.0/toolbox/news-case-study.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+},
+```
+
+If `IntersectionObserver` isn't available, the application briefly delays displaying the ad to let the rest of the page load first.
+
+## AMP version of the News app
+
+We built a [version of the News app](https://github.com/PolymerLabs/news/tree/amp) for integration with [AMP (Accelerated Mobile Pages)](https://www.ampproject.org/) documents. In the AMP version of News, the HTML content files (for example, `/data/articles/it-takes-teacher.html`) are replaced by AMP documents which can be loaded as standalone pages.
+
+The AMP project supports adding an AMP document to a shadow root on your page. Here’s an example: <code>[https://github.com/ampproject/amphtml/blob/master/examples/pwa/pwa.js](https://github.com/ampproject/amphtml/blob/master/examples/pwa/pwa.js)</code>
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: If you're not going to put a period at the end of this link (which would probably look funky), I'd put it on a separate line.

File: app/1.0/toolbox/news-case-study.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+
+`news-article.html`
+```html
+<div class="container" fade-in$="[[!loading]]" hidden$="[[failure]]">
+  <article class="flex">
+    <amp-viewer id="ampViewer" src="[[_getArticleSrc(article.id)]]"></amp-viewer>
+  </article>
+```
+
+Note: the AMP version of the app does not support swiping between articles - doing so would require loading the title, image and author of the previous or next article. In the AMP version of News, that data is only loaded when the article content is loaded. It is not loaded with the list of articles.
```
Comment (arthurevans) [inline app/1.0/toolbox/news-case-study.md]: No space around em dashes.

File: app/1.0/toolbox/news-case-study.md [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+Views in the News app use the same implementation as the Shop app. [See the Shop app case study for 
+more information on views and 
+routing](https://www.polymer-project.org/1.0/toolbox/case-study#views).
+
+For more information on encapsulated routing, see [Encapsulated routing with 
+elements](https://www.polymer-project.org/1.0/blog/routing).
+
+For more information on the `<app-route>` Polymer element used in these implementations, see the 
+[`<app-route>` API reference](https://elements.polymer-project.org/elements/app-route).
```
Comment (abdonrd) [inline app/1.0/toolbox/news-case-study.md]: I suggest update to:
`https://www.webcomponents.org/element/PolymerElements/app-route`

File: app/1.0/toolbox/news-case-study.md [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+<news-network-warning
+    hidden$="[[!failure]]"
+    offline="[[offline]]"
+    on-try-reconnect="_tryReconnect"></news-network-warning>
+```
+
+If the user is offline but category data is cached and ServiceWorker is installed,
+the cached content is displayed.
+
+-   [`<app-route>` API reference](https://elements.polymer-project.org/elements/app-route)
```
Comment (abdonrd) [inline app/1.0/toolbox/news-case-study.md]: I suggest update to:
`https://www.webcomponents.org/element/PolymerElements/app-route`

Comment (katejeffreys) [inline app/1.0/toolbox/news-case-study.md]: Thank you for catching these!

Comment (abdonrd) [inline app/1.0/toolbox/news-case-study.md]: Your welcome! 😄

**Comment by arthurevans** [comment]
@katejeffreys one more change required here—the app/1.0/nav.yaml file needs to be updated to show the news case study. You can copy the three lines here for the Shop case study:

https://github.com/Polymer/docs/blob/master/app/1.0/nav.yaml#L229

(title, path and indent)

**Comment by arthurevans** [comment]
@katejeffreys I'm going to push a few changes to this branch and then merge it.

**Pull request merged**